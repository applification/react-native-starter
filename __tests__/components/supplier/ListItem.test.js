import React from 'react';
import renderer from 'react-test-renderer';

import SupplierListItem from '../../../src/components/supplier/ListItem';

const SUPPLIER_ITEM_DATA = {
  gender: 'male',
  name: {
    title: 'mr',
    first: 'mason',
    last: 'walker'
  },
  location: {
    street: '9366 bairds road',
    city: 'upper hutt',
    state: 'west coast',
    postcode: 49293
  },
  email: 'mason.walker@example.com',
  login: {
    username: 'whitemouse414',
    password: 'lowell',
    salt: 'I1x0ZEtp',
    md5: '6167a2c479522db9f7befcc832e7ace8',
    sha1: '94e8cc37ceaaba2450cb97fe3e1f52d3b1ea5ab2',
    sha256: 'dbbebda09b08fb5d79073c637a799066c3f0b9dd7c09ac59f6189483fd6b6214'
  },
  dob: '1974-09-20 09:28:59',
  registered: '2007-02-24 02:51:20',
  phone: '(907)-186-0638',
  cell: '(737)-135-2392',
  id: {
    name: '',
    value: null
  },
  picture: {
    large: 'https://randomuser.me/api/portraits/men/84.jpg',
    medium: 'https://randomuser.me/api/portraits/med/men/84.jpg',
    thumbnail: 'https://randomuser.me/api/portraits/thumb/men/84.jpg'
  },
  nat: 'NZ'
};

describe('SUPPLIER LIST ITEM COMPONENT', () => {
  it('renders a Supplier List Item Compoment', () => {
    expect(
      renderer.create(
        <SupplierListItem data={SUPPLIER_ITEM_DATA} nav={{ navigate: null }} />
      )
    ).toMatchSnapshot();
  });
});
