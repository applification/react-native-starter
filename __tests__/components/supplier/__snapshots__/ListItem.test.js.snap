// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SUPPLIER LIST ITEM COMPONENT renders a Supplier List Item Compoment 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    avatar={true}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 15,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 0,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <Image
        source={
          Object {
            "uri": "https://randomuser.me/api/portraits/thumb/men/84.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 50,
            "width": 50,
          }
        }
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        Mason Walker
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        9366 bairds road upper hutt west coast 49293
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "alignSelf": null,
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "200",
          }
        }
        virtual={undefined}
      >
        (737)-135-2392
      </Text>
    </View>
  </View>
</View>
`;
